pontoon/base/static/js/themeSwitcher.js

```diff
@@ -0,0 +1,77 @@
+$(function () {
+  function getSystemTheme() {
+    if (
+      window.matchMedia &&
+      window.matchMedia('(prefers-color-scheme: dark)').matches
+    ) {
+      return 'dark';
+    } else {
+      return 'light';
+    }
+  }
+
+  function updateTheme(newTheme) {
+    if (newTheme === 'system') {
+      newTheme = getSystemTheme();
+    }
+    $('body')
+      .removeClass('dark-theme light-theme system-theme')
+      .addClass(`${newTheme}-theme`);
+
+    if (newTheme !== 'system') {
+      Pontoon.endLoader(`Theme changed to ${newTheme}.`);
+    }
+  }
+
+  window
+    .matchMedia('(prefers-color-scheme: dark)')
+    .addEventListener('change', function (e) {
+      if ($('.appearance .toggle-button button.system').hasClass('active')) {
+        if (e.matches) {
+          updateTheme('dark');
+        } else {
+          updateTheme('light');
+        }
+      }
+    });
+
+  $(document).ready(function () {
```

Comment:
We are already in the `$(document).ready()` as of the first line of this file, so you can remove this wrapper:
https://learn.jquery.com/using-jquery-core/document-ready/